### ISSUE TITLE
Nested table without a key is not delimited

### ISSUE THREAD

**Issue by whc2001** [CXuesong/Luaon.NET#4]
It's me again, lol.

If a nested table does not have a key (array-like member), then it is not delimited from the last token.
```
            LTable outer = new LTable();
            LTable inner1 = new LTable();
            LTable inner2 = new LTable();

            outer.Add(1);
            outer.Add(inner1);
            outer.Add("A");

            inner2["test1"] = 1;
            inner2["test2"] = 2;

            inner1.Add(4);
            inner1.Add(5);
            inner1.Add(inner2);

            Console.WriteLine(outer.ToString());
```
![I~9QC$V0P7R7VT%DQSP9KK4](https://user-images.githubusercontent.com/16266909/150704873-4e1f9efb-eb6d-4fe2-8310-40d03182bd85.png)

However if it has a key (dict-like member), then it seems fine.
```
            LTable outer = new LTable();
            LTable inner1 = new LTable();
            LTable inner2 = new LTable();

            outer.Add(1);
            outer.Add(inner1);
            outer.Add("A");

            inner2["test1"] = 1;
            inner2["test2"] = 2;

            inner1.Add(4);
            inner1.Add(5);
            inner1.Add("testkey", inner2);    // <- Add a key

            Console.WriteLine(outer.ToString());
```
![IAU HMIO_FSTKKBSA3EUY5C](https://user-images.githubusercontent.com/16266909/150704949-7b1aac1d-1a15-4d19-9e0d-2b04f2e1b277.png)

Again after some trial-and-error debugging, seems like `WriteStartTable()` in `LuaTableTextWriter.cs` is not designed to delimit the last value as it simply goes to `TableStart` state.

https://github.com/CXuesong/Luaon.NET/blob/0ce291b4b5b62911ebffbc5a002d4465a21254b3/Luaon.NET/LuaTableTextWriter.cs#L208-L216

If I change line 212 to `DelimitLastValue(Token.TableStart);`, the problem seems to be solved.

![image](https://user-images.githubusercontent.com/16266909/150705140-a9521c30-a90a-4a6f-8a3c-346ccdde7f9d.png)

**Comment by CXuesong** [comment]
I do think we might need some UT, but I don't think I'll devote sufficient time on this yet 😂

Let's just fix what we've found right now: [`v0.2.5`](https://github.com/CXuesong/Luaon.NET/releases/tag/v0.2.5).

**Comment by whc2001** [comment]
One of my friend's team has written a Python version of LuaTable parser, and there are some UTs in that repo. Maybe I'll try to rewrite them with C# later.

https://github.com/leafvmaple/luadata/blob/master/luadata/serializer/__test__.py
